shell/platform/linux/fl_value.cc

```diff
@@ -232,6 +239,10 @@ static void value_to_string(FlValue* value, GString* buffer) {
       g_string_append(buffer, "}");
       return;
     }
+    case FL_VALUE_TYPE_CUSTOM:
```

Comment:
Yes, this is nested - the custom FlValue contains both the type (an integer) and the value (a pointer).

Reply:
Is the doubling of types just to keep the return type of `fl_value_get_type`? I'm a concerned that having two different type IDs is going to be confusing; if we soft-deprecated `fl_value_get_type` in favor of a `fl_value_get_extensible_type` (or similar) that returned an int instead of an enum, could we collapse to a single type?

Reply:
The reason I didn't replace the type with the ID was:
- Unsure if the ID for the basic types (i.e. < 128) should be exposed in the API.
- If we replace an enum with an ID then we lose compiler checks that you've handled all types correctly.
- There's a getter for each enum value - all custom types are stored as a pointer. If we use IDs as a key then some IDs (the primitive types) have a getter and many IDs use the pointer getter.
- I expect most systems won't need to support custom types, so the simple case is still simple (i.e. just use the existing enum and stub out the custom case), and the more complex case is doable (i.e. handle custom types by overriding the virtual method).

Reply:
Fair enough. The C implementation is in sort of an odd space between a more type-generic case like ObjC, and the strict type system of the C++ version, so there's no clearly correct answer, and those arguments make sense to me.

Reply:
Sadly yes :)